core/src/main/scala-3/spire/optional/unicode.scala

```diff
@@ -0,0 +1,90 @@
+package spire
+package optional
+
+import spire.algebra._
+import spire.algebra.lattice._
+import spire.math._
+
+package object unicode {
```

Comment:
I think Scala 3 deprecated package objects, and since this is a Scala 3 only source, maybe just `object` suffices.

Reply:
Another possibility: we can just put this file inside the `unicode` package and make these top-level definitions since Scala 3 supports that as well. No reason these should be inside an object :)

Reply:
Right, that sounds a good choice, I'll update

Reply:
Fixed, and I added a tiny sanity check test